crates/oxc_linter/src/rules/no_mixed_operators.rs

```diff
@@ -0,0 +1,344 @@
+use oxc_ast::{AstKind, GetSpan, Span};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::AstNode;
+
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-mixed-operators): Disallow mixed binary operators")]
+#[diagnostic()]
+struct NoMixedOperatorsDiagnostic(
+    &'static str, /*Node Operator */
+    &'static str, /*Parent Operator */
+    #[label(
+        "Unexpected mix of {0} and {1}. Use parentheses to clarify the intended order of operations"
+    )]
+    pub Span, /*Span of the operator */
+);
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct NoMixedOperators {
+    /// Disallow Mixed operators within one group.
+    groups: Vec<Vec<&'static str>>,
+    /// Allow operators of the same precedence to be mixed.
+    allow_same_precedence: bool,
+}
+
+declare_oxc_lint! {
+  /// ### What it does
+  /// Disallow mixed binary operators.
+  ///
+  /// ### Why is this bad?
+  /// Enclosing complex expressions by parentheses clarifies the developer’s intention,
+  /// which makes the code more readable. This rule warns when different operators
+  /// are used consecutively without parentheses in an expression.
+  ///
+  /// ### Examples
+  /// ```javascript
+  /// var foo = a && b || c || d;    /*BAD: Unexpected mix of '&&' and '||'.*/
+  /// var foo = (a && b) || c || d;  /*GOOD*/
+  /// var foo = a && (b || c || d);  /*GOOD*/
+  /// ```
+  NoMixedOperators,
+  correctness
+}
+
+impl Rule for NoMixedOperators {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let node_kind = node.get().kind();
+        if !matches!(node_kind, AstKind::BinaryExpression(_) | AstKind::LogicalExpression(_)) {
+            return;
+        }
+
+        let parent = ctx.parent_node(node);
+        if parent.is_none() {
+            return;
+        }
+        let parent = parent.unwrap();
+        let parent_kind = parent.get().kind();
```

Comment:
You can you `ctx.parent_kind(node)` if don't need the know. You may also add more handy utils to `LintContext`.

Reply:
Good point! Done.